packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts

```diff
@@ -71,22 +71,25 @@ export function useUploadFiles({
       };
 
       if (file) {
-        resolveFile({ processFile, file, key }).then(({ key, file }) => {
-          onUploadStart?.({ key });
-          const uploadTask = uploadFile({
-            file,
-            key,
-            level: accessLevel,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            completeCallback: onComplete,
-          });
+        resolveFile({ processFile, file, key }).then(
+          ({ key, file, ...rest }) => {
+            onUploadStart?.({ key });
+            const uploadTask = uploadFile({
+              ...rest,
+              file,
+              key,
+              level: accessLevel,
+              progressCallback: onProgress,
+              errorCallback: onError,
+              completeCallback: onComplete,
+            });
```

Comment:
For readability, can this be:
```suggestion
          ({ key: processedKey, file, ...rest }) => {
            onUploadStart?.({ key: processedKey });
            const uploadTask = uploadFile({
              ...rest,
              key: processedKey,
              level: accessLevel,
              progressCallback: onProgress,
              errorCallback: onError,
              completeCallback: onComplete,
            });
```